src/core/diags/factory_diags.cpp

```diff
@@ -221,7 +221,24 @@ Error Diags::ProcessFrame(uint8_t aArgsLength, char *aArgs[])
     Error    error = kErrorNone;
     uint16_t size  = OT_RADIO_FRAME_MAX_SIZE;
 
-    VerifyOrExit(aArgsLength == 1, error = kErrorInvalidArgs);
+    memset(&mTxPacket->mInfo, 0, sizeof(mTxPacket->mInfo));
+
+    while (aArgsLength > 1)
```

Comment:
Do we need `while()` here?
We can allow a single `-s` option. Any benefit to allow multiple

Reply:
I'm going to add flags for other fields in txinfo like csma/ca. Do you think I should add `while` in that PR?

Reply:
Yes please. Make sense to add it then.